beacon_chain/state/helpers.py

```diff
@@ -131,3 +134,26 @@ def get_new_shuffling(seed,
             committee=indices
         ) for j, indices in enumerate(shard_indices)])
     return o
+
+
+def get_indices_for_slot(
+        crystallized_state,
+        slot,
+        config=DEFAULT_CONFIG):
+    cycle_length = config['cycle_length']
+
+    ifh_start = crystallized_state.last_state_recalc - cycle_length
+    assert ifh_start <= slot < ifh_start + cycle_length * 2
+    return crystallized_state.indices_for_slots[slot - ifh_start]
+
+
+def get_block_hash(
+        active_state,
+        current_block,
+        slot,
+        config=DEFAULT_CONFIG):
+    cycle_length = config['cycle_length']
+
+    sback = current_block.slot_number - cycle_length * 2
+    assert sback <= slot < sback + cycle_length * 2
+    return active_state.recent_block_hashes[slot - sback]
```

Comment:
have you considered the case where `sback` could be negative?
Example:
`current_block.slot_number` = 10
`sback` = 10 - 64 * 2 = -118
then it'll return `active_state.recent_block_hashes[slot + 118]`

Reply:
I think it's okay since practically:
1. `active_state.recent_block_hashes` was initialized to a list of `CYCLE_LENGTH * 2` zero hashes, and then updated via `get_new_recent_block_hashes` block by block. 
2. The domain of `slot - sback` was guaranteed by the assertion `sback <= slot < sback + cycle_length * 2`, that said, `0 <= slot - sback < CYCLE_LENGTH * 2`.

I will add a test of the scenario you mentioned, thank you! 👍